lib/iris/tests/integration/test_pickle.py

```diff
@@ -31,57 +32,82 @@
     from iris_grib.message import GribMessage
 
 
-@tests.skip_data
-@tests.skip_grib
-class TestGribMessage(tests.IrisTest):
-    def test(self):
-        # Check that a GribMessage pickles without errors.
-        path = tests.get_data_path(('GRIB', 'fp_units', 'hours.grib2'))
-        messages = GribMessage.messages_from_filename(path)
-        message = next(messages)
+class Common(object):
+    def pickle_cube(self, protocol):
+        # Ensure that data proxies are pickleable.
+        cube = iris.load(self.path)[0]
         with self.temp_filename('.pkl') as filename:
             with open(filename, 'wb') as f:
-                pickle.dump(message, f)
+                pickle.dump(cube, f, protocol)
+            with open(filename, 'rb') as f:
+                ncube = pickle.load(f)
+        self.assertEqual(ncube, cube)
 
-    def test_data(self):
-        # Check that GribMessage.data pickles without errors.
-        path = tests.get_data_path(('GRIB', 'fp_units', 'hours.grib2'))
-        messages = GribMessage.messages_from_filename(path)
-        message = next(messages)
-        with self.temp_filename('.pkl') as filename:
-            with open(filename, 'wb') as f:
-                pickle.dump(message.data, f)
+    def test_protocol_0(self):
+        self.pickle_cube(0)
 
+    def test_protocol_1(self):
+        self.pickle_cube(1)
 
-class Common(object):
-    # Ensure that data proxies are pickleable.
-    def pickle_cube(self, path):
-        cube = iris.load(path)[0]
+    def test_protocol_2(self):
+        self.pickle_cube(2)
+
+
+@tests.skip_data
+@tests.skip_grib
+class TestGribMessage(Common, tests.IrisTest):
+    def setUp(self):
+        self.path = tests.get_data_path(('GRIB', 'fp_units', 'hours.grib2'))
+
+    def pickle_obj(self, obj):
         with self.temp_filename('.pkl') as filename:
             with open(filename, 'wb') as f:
-                pickle.dump(cube, f)
+                pickle.dump(obj, f)
+
+    # These probably "ought" to work, but currently fail.
+    # see https://github.com/SciTools/iris/pull/2608
+    @unittest.expectedFailure
+    def test_protocol_0(self):
+        super(TestGribMessage, self).test_protocol_0()
```

Comment:
@cpelley Now we're exclusively Python3, isn't this now simply `super().test_protocol_0()`?

If true, also applies through out. Fix this and I'll merge 😄

Reply:
@cpelley This'll all be resolved by https://github.com/SciTools/iris/issues/3457, so let's bank some progress :+1: